Add Playwright composite action to reduce workflow duplication

snomiao opened the pull request:
This PR introduces a reusable composite action for Playwright setup to reduce duplication across workflows.

## Changes
- Created `.github/actions/setup-playwright/action.yml` composite action that:
  - Detects or uses provided Playwright version
  - Caches Playwright browsers with intelligent cache keys
  - Installs browsers only when cache miss occurs
  - Installs OS dependencies when cache hit occurs
  
## Technical Details
- **Important:** The composite action requires `shell: bash` for all `run` steps as per [GitHub Actions requirements for composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file). This is a mandatory field for composite actions, unlike regular workflow steps.
- Updated workflow paths to account for repository checkout locations (some workflows checkout to subdirectories like `ComfyUI_frontend/`)
- Uses conditional caching to avoid redundant browser installations

## Benefits
- Reduces code duplication across 6 workflow files
- Centralizes Playwright caching logic  
- Consistent browser setup across all workflows
- Easier maintenance and updates
- Faster CI runs through intelligent caching

## Affected Workflows
- `.github/workflows/test-ui.yaml` (2 uses)
- `.github/workflows/i18n-custom-nodes.yaml`
- `.github/workflows/i18n-node-defs.yaml`
- `.github/workflows/i18n.yaml`
- `.github/workflows/test-browser-exp.yaml`

snomiao committed:
[feat] Add Playwright composite action to reduce workflow duplication

This PR introduces a reusable composite action for Playwright setup to reduce duplication across workflows.

- Created .github/actions/setup-playwright/action.yml composite action that:
  - Detects or uses provided Playwright version
  - Caches Playwright browsers with intelligent cache keys
  - Installs browsers only when cache miss occurs
  - Installs OS dependencies when cache hit occurs

- The composite action requires 'shell: bash' for all run steps as per GitHub Actions requirements for composite actions
- Updated workflow paths to account for repository checkout locations (some workflows checkout to subdirectories)

- Reduces code duplication across 6 workflow files
- Centralizes Playwright caching logic
- Consistent browser setup across all workflows
- Easier maintenance and updates

snomiao commented on .github/actions/setup-playwright/action.yml:
```diff
@@ -0,0 +1,50 @@
+  steps:
+    - name: Detect Playwright version
+      id: detect-version
+      shell: bash
```
TODO: no need to specify a shell here

snomiao replied:
- **Important:** The composite action requires `shell: bash` for all `run` steps as per [GitHub Actions requirements for composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file). This is a mandatory field for composite actions, unlike regular workflow steps.

DrJKL reviewed (changes requested):
I love it, but I think it could be simplified a bit further.

snomiao committed:
Simplify Playwright composite action per review feedback

- Remove playwright-version input (always detect from package.json)
- Remove cache-key-suffix input
- Remove playwright-version output (not used by callers)
- Update all workflow files to remove these parameters

snomiao committed:
Remove cache-key-suffix from workflow files per review feedback

snomiao committed:
Remove working-directory parameter from composite action

Since it's always 'ComfyUI_frontend', hardcode it in the action itself per review feedback

christian-byrne commented:
There are some conflicts with main

snomiao committed:
Merge origin/main and resolve conflicts

- Resolved conflict in update-playwright-expectations.yaml to use new composite action

actions-user committed:
[auto-fix] Apply ESLint and Prettier fixes

snomiao commented on .github/workflows/tests-ci.yaml:
```diff
@@ -123,22 +116,8 @@ jobs:
-        run: pnpm exec playwright install-deps
-        working-directory: ComfyUI_frontend
+      - name: Setup Playwright
+        uses: ./ComfyUI_frontend/.github/actions/setup-playwright
```
Agree! maybe we can simplify those things to setup-frontend-and-playwright.yaml in near future so the working directories would look nicer
